packages/opentelemetry-core/src/common/time.ts

```diff
@@ -181,3 +172,14 @@ export function isTimeInput(value: unknown): value is api.HrTime | number | Date
     value instanceof Date
   );
 }
+
+export function addHrTimes(time1: api.HrTime, time2: api.HrTime): api.HrTime {
+  const out: api.HrTime = [time1[0] + time2[0], time1[1] + time2[1]];
+
+  while(out[1] > SECOND_TO_NANOSECONDS) {
```

Comment:
Any performance concerns about how many times this while loop has to run while correcting time drift, since every second of time drift causes an extra run of this loop (eg. 8 hours of drift -> 28800 loops; 24 hours of drift -> 86400 loops)?

manually measuring this am getting <0.1ms for 6h of drift, 0.2-0.3ms for 30h of drift [varies depending on device ofc]

Reply:
I was thinking the same thing. Changed to use a division and remainder